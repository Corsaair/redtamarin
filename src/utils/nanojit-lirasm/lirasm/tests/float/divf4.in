; This Source Code Form is subject to the terms of the Mozilla Public
; License, v. 2.0. If a copy of the MPL was not distributed with this
; file, You can obtain one at http://mozilla.org/MPL/2.0/.

; Try to exercise as many different possibilities for the register allocator as
; is feasible.

p1 = allocp 16
p2 = allocp 16

d1a = immf4 1.5 -1.5 0.0 1
d1 = divf4 d1a d1a ; X = X / X
stf4 d1 p1 0       ; 1,1,-1.#IND,1

d2a = immf4 4.0 -8.0 4.0 -8.0
d2b = immf4 8.0 4.0 -8.0 -4.0
d2 = divf4 d2a d2b ; X = X / Y
stf4 d2b p2 0
stf4 d2 p2 0   ;  0.5,-2,-0.5,2

d3a = ldf4 p1 0
d3b = ldf4 p2 0
d3 = divf4 d3a d3b ; X = Y / X
stf4 d3a p2 0
stf4 d3 p2 0    ; 2,-0.5,-1.#IND,0.5

f1 = immf -1
f2 = immf -0
inf = divf f1 f2
stf inf p1 8
stf f1 p2 8

d4a = ldf4 p2 0
d4b = ldf4 p1 0
d4 = divf4 d4a d4b ; X = Y / Z
stf4 d4a p1 0
stf4 d4b p2 0
stf4 d4 p1 0  ; 2,-0.5,-0,0.5

d = ldf4 p1 0
retf4 d
