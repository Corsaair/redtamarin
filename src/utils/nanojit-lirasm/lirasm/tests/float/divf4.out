Output is: 2,-0.5,-0,0.5
